### PR TITLE
Add arg_names attribute to FunctionContext and MethodContext

### PR DESCRIPTION
**Pull request by mkurnikov**
References https://github.com/python/mypy/issues/5409.

### CODE REVIEW

**Comment by mkurnikov** [comment]
I'll finish it today or on the weekend. With the recent refactoring, it will be even easier.

**Comment by mkurnikov** [comment]
WIP, missing tests for `arg_kinds`, existing tests should be rewritten and docs for new context attributes. Will try to finish on this weekend.

Explanation of `callee_arg_names`: 
Guido mentioned in the 
https://github.com/python/mypy/issues/5409#issue-346398965
> I needed the names of the arguments but they were not directly available. I ended up using rtype.type.names['\_\_init__'].node.arg_names (with various type and None guards) and subtracting 1 to compensate for self, which looked very hacky. It would be nice if FunctionContext also has an arg_names argument.

I've encountered the same thing in `django-stubs`. 

Five arrays of the same length don't look great, but I couldn't find any other way.

File: mypy/plugin.py [hunk 1]
```diff
@@ -159,7 +162,10 @@ def qualified_name(self, n: str) -> str:
...
     'MethodContext', [
         ('type', Type),                    # Base object type for method call
-        ('arg_types', List[List[Type]]),
+        ('callee_arg_names', List[Optional[str]]),
+        ('arg_names', List[List[Optional[str]]]),
+        ('arg_kinds', List[List[int]]),
+        ('arg_types', List[List[Type]]),  # List of actual caller types for each formal argument
         ('default_return_type', Type),
         ('args', List[List[Expression]]),
         ('context', Context),
```
Comment (ilevkivskyi) [inline mypy/plugin.py]: Could you please also add a comment for every existing attribute here and above?

**Comment by mkurnikov** [comment]
I'll finish it tomorrow.

**Comment by ilevkivskyi** [comment]
Great, thanks!

**Comment by ilevkivskyi** [comment]
Also please never rebase, it is much harder to review, and GitHub only sends a notification when you push a new commit.

**Comment by ilevkivskyi** [comment]
Thanks for the PR!

**Comment by mkurnikov** [comment]
Got it.

**Pull request merged**